src/types.rs

```diff
@@ -0,0 +1,6 @@
+use aws_sdk_s3::primitives::SdkBody;
+use aws_smithy_http::result::SdkError;
+use http::Response;
+
+/// Convenience wrapper to handle http response
+pub type DefaultSdkError<E> = SdkError<E, Response<SdkBody>>;
```

Comment:
**question:** Will this all still work if we make the type `pub(crate)` rather than `pub`? The reason I ask is, if we can get away with making it `pub(crate)` then is means that this change should be invisible to consumers; if we need to have the type `pub` then it means we're exposing this type in our public API and we'll have to think a bit about what consumers should do when they upgrade.

```suggestion
pub(crate) type DefaultSdkError<E> = SdkError<E, Response<SdkBody>>;
```

Reply:
**suggestion:** Would we ever expect someone to provide a *different* version of this type when working on or using this codebase? If not, then having "default" in the name of the type here may risk confusion (as they wonder what non-default options might exist) and just naming it `SdkError<E>` would be simpler. It's OK to shadow the name of the type from `aws_smithy_http` if we're going to use it for more-or-less the same purpose.